OpenRA.Game/Graphics/CursorProvider.cs

```diff
@@ -26,21 +27,15 @@ public CursorProvider(ModData modData)
 			var sequenceYaml = MiniYaml.Merge(modData.Manifest.Cursors.Select(
 				s => MiniYaml.FromStream(fileSystem.Open(s), s)));
 
-			var shadowIndex = new int[] { };
-
 			var nodesDict = new MiniYaml(null, sequenceYaml).ToDictionary();
-			if (nodesDict.ContainsKey("ShadowIndex"))
-			{
-				Array.Resize(ref shadowIndex, shadowIndex.Length + 1);
-				Exts.TryParseIntegerInvariant(nodesDict["ShadowIndex"].Value,
-					out shadowIndex[shadowIndex.Length - 1]);
-			}
 
-			var palettes = new Dictionary<string, ImmutablePalette>();
-			foreach (var p in nodesDict["Palettes"].Nodes)
-				palettes.Add(p.Key, new ImmutablePalette(fileSystem.Open(p.Value.Value), shadowIndex));
+			var pals = modData.DefaultRules.Actors["world"].TraitInfos<ICursorPaletteInfo>()
+				.ToDictionary(p => p.Palette, p => p);
```

Comment:
Unfortunately this crashes in all mods because we have multiple definitions for the `terrain` palette, with the correct one selected when loading the map.  We don't know enough to select the one we want this early in the game, so we're going to need to adjust the plumbing to account for this.

I think the cleanest approach might be to add a `public bool CursorPalette = false;` to the traits that implement this interface, and have them return a `null` palette name if the cursor is not explicitly enabled.
We can then `.Select(p => p != null)` here to drop them, with an extra bonus of not `ReadPalette`ing a bunch of things we aren't going to use.